### PR TITLE
Feature validation not more then 5

### PR DESCRIPTION
**Pull request by elBeato**
## Issue: 
#169

### CODE REVIEW

File: src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java [hunk 1]
```diff
@@ -38,48 +43,73 @@ void setup() {
 
   @Test
   void player_join_game_valid_game_state() {
-    PARTICIPATION_1.setParticipationNumber(0);
+    PARTICIPATION.setParticipationNumber(0);
```
Comment (salwil) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java:46]: Is there a reason for sequence number 0,1,3? Why not 1,2,3 or 0,1,2?

File: src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java [hunk 2]
```diff
@@ -6,19 +6,24 @@
...
+import java.util.stream.Stream;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
 import org.springframework.web.client.HttpClientErrorException;
 
 class ParticipationValidatorTest {
+  private static final int MAX_NUMBER_OF_PLAYERS = 5;
+
```
Comment (salwil) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java:19]: Detail: to me the test class would have more impression of reliability, if we instantiated the game in the BeforeEach section every time new. So no old values are carried over to from one test method to the next.

File: src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java [hunk 3]
```diff
@@ -38,48 +43,73 @@ void setup() {
...
+    GAME_FINDING_PLAYERS.setParticipation(
+        Stream.generate(Participation::new)
+            .limit(MAX_NUMBER_OF_PLAYERS)
+            .collect(Collectors.toList()));
 
     assertThrows(
         HttpClientErrorException.class,
-        () -> participationValidator.onUpdateParticipation(PARTICIPATION_1));
+        () -> participationValidator.onBeforeCreateParticipation(PARTICIPATION));
```
Comment (salwil) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java:113]: I like the minimalistic way of how you test this :D

File: src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java [hunk 4]
```diff
@@ -38,48 +43,73 @@ void setup() {
...
+    PARTICIPATION.setParticipationNumber(0);
+    PARTICIPATION.setGame(GAME_ILLEGAL_STATE);
+
+    assertThrows(
+        HttpClientErrorException.class,
+        () -> participationValidator.onBeforeCreateParticipation(PARTICIPATION));
+  }
+
+  @Test
+  void game_has_space_for_one_additional_players() {
```
Comment (BacLuc) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java:92]:
```suggestion
  void game_has_space_for_one_additional_player() {
```

File: src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java [hunk 5]
```diff
@@ -29,7 +34,7 @@ void setup() {
     GAME_FINDING_PLAYERS.setId(1L);
     GAME_ILLEGAL_STATE.setId(2L);
 
-    PARTICIPATION_1.setId(1L);
+    PARTICIPATION.setId(1L);
```
[the remark this answers is not in the document]
Comment (elBeato) [reply src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java:37]: Das hani gmacht um variablen zu sparen. Dann dann kann ich die PARTICIPATION in mehreren tests verwenden und erst, ohne zahl ist immer 1 und sonst steht die zahl.

File: src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java (hunk 1 above)
Comment (elBeato) [reply src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java:46]: Nein, ist random, dürfte ich wahrscheinlich auch weglassen.

File: src/main/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/entity/Game.java [hunk 6]
```diff
@@ -81,6 +81,11 @@ public Collection<Participation> getParticipations() {
     return participations;
   }
 
+  @JsonIgnore
+  public void setParticipation(Collection<Participation> participations) {
```
[the remark this answers is not in the document]
Comment (elBeato) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/entity/Game.java:85]: ah shit...damn right. I know this short-cuts, but somtimes I like to write. :-)

File: src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java [hunk 7]
```diff
@@ -6,19 +6,24 @@
...
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
 import org.springframework.web.client.HttpClientErrorException;
 
 class ParticipationValidatorTest {
+  private static final int MAX_NUMBER_OF_PLAYERS = 5;
```
[the remark this answers is not in the document]
Comment (elBeato) [reply src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java:18]: Yes I had the same idea, may I would create a ENUM, STATIC class which all corner points variables - like this - are saved and reuse from that class or whatever...what you think?

[the remark this answers is not in the document]
Comment (BacLuc) [reply src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/ParticipationValidatorTest.java]: i would just make the constant available from the class and the unit test.
I vote against a god class with all constants

**Review by BacLuc — APPROVED** [review]

**Pull request merged**